src/rpc/methods/eth/filter/mod.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod event;
+mod store;
+mod tipset;
+
+use crate::rpc::eth::filter::event::*;
+use crate::rpc::eth::filter::tipset::*;
+use crate::rpc::eth::types::*;
+use crate::shim::address::Address;
+use crate::shim::clock::ChainEpoch;
+use crate::utils::misc::env::env_or_default;
+use ahash::AHashMap as HashMap;
+use cid::Cid;
+use serde::*;
+use std::sync::Arc;
+use store::*;
+use thiserror::Error;
+
+#[derive(Error, Debug, Clone)]
+pub enum FilterError {
+    #[error("filter already registered")]
+    AlreadyRegistered,
+    #[error("filter not found")]
+    NotFound,
+    #[error("maximum number of filters registered")]
+    MaxFilters,
+    #[error("uuid generation error")]
+    UuidError(#[from] uuid::Error),
+    #[error("Not supported")]
+    NotSupported,
+    #[error("Parsing error: {0}")]
+    ParsingError(String),
+    #[error("Installation error: {0}")]
+    InstallationError(String),
+    #[error("Removal error: {0}")]
+    RemovalError(String),
+}
+
+pub struct EthEventHandler {
+    filter_store: Option<Arc<dyn FilterStore>>,
+    max_filter_height_range: ChainEpoch,
+    event_filter_manager: Option<Arc<EventFilterManager>>,
+    tipset_filter_manager: Option<Arc<TipSetFilterManager>>,
+}
+
+impl EthEventHandler {
+    pub fn new() -> Self {
+        let max_filters: usize = env_or_default("MAX_FILTERS", 100);
+        let max_filter_results: usize = env_or_default("MAX_FILTER_RESULTS", 10000);
+        let max_filter_height_range: i64 = env_or_default("MAX_FILTER_HEIGHT_RANGE", 2880);
```

Comment:
Are those environments variable for testing purposes or for end-users? In both cases, it would be great to have some documentation.

Reply:
added